pkg/crd/gen.go

```diff
@@ -76,16 +89,45 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 		return nil
 	}
 
+	crdVersions := g.CRDVersions
+
+	if len(crdVersions) == 0 {
+		// put v1beta1 first to make it the "default" file for compatibility purposes
+		crdVersions = []string{"v1beta1", "v1"}
+	}
+
 	for _, groupKind := range kubeKinds {
 		parser.NeedCRDFor(groupKind, g.MaxDescLen)
-		crd := parser.CustomResourceDefinitions[groupKind]
+		crdRaw := parser.CustomResourceDefinitions[groupKind]
+		addAttribution(&crdRaw)
+
+		versionedCRDs := make([]interface{}, len(crdVersions))
+		for i, ver := range crdVersions {
+			conv, err := AsVersion(crdRaw, schema.GroupVersion{Group: apiext.SchemeGroupVersion.Group, Version: ver})
```

Comment:
Summarizing our offline conversation --> so if I am using v1 specific feature (say defaulting), generated v1beta1 CRD will have defaulting specific features enabled, but when submitted to API-Server will result an error.
So current KB projects will run in to this issue when they start using v1 features. Still wondering if there is a better way to handle this transition.